### PR TITLE
Changes to allow for easier testing

### PR DESCRIPTION
**Pull request by George3d6**
* Changed the `test` method to allow passing the full `.explanation` property to the scoring function (in order allow for e.g. evaluating numerical ranges or evaluation based on the actual predictions + confidences)
* Change `learn` in order to return the validation dataset

Note: Not to be merged just yet, but feel free to review if you have the time.

### CODE REVIEW

**Comment by btseytlin** [comment]
How about we just return the whole transaction? Then we can inspect the validation df and other info too

Also this Pr is targeting stable. Don't know if intended!

**Comment by George3d6** [comment]
@btseytlin not intended.

Also, returning the whole transaction or making the transaction a visible property of the predictor might indeed be better.

I think, for now, I will also add the `validate` method based on our discussion.

**Review by btseytlin — APPROVED** [review]

**Review by paxcema — APPROVED** [review]

File: mindsdb_native/libs/controllers/predictor.py [hunk 1]
```diff
@@ -242,10 +243,11 @@ def learn(self,
                 if old_hmd['from_data'] is not None:
                     heavy_transaction_metadata['from_data'] = old_hmd['from_data']
 
-            LearnTransaction(session=self,
+            self.transaction = transaction = LearnTransaction(session=self,
```
Comment (maximlopin) [inline mindsdb_native/libs/controllers/predictor.py]: typo (`self.transaction = transaction = ...`)

**Review by maximlopin — APPROVED** [review]

File: mindsdb_native/libs/controllers/predictor.py (hunk 1 above)
Comment (George3d6) [reply mindsdb_native/libs/controllers/predictor.py]: Fixed :)

**Pull request merged**